src/utils/parser.ts

```diff
@@ -0,0 +1,35 @@
+import { registerSchemaParser, parse } from '@asyncapi/parser';
+import { Request } from 'express';
+
+import ramlDtParser from '@asyncapi/raml-dt-schema-parser';
+import openapiSchemaParser from '@asyncapi/openapi-schema-parser';
+import avroSchemaParser from '@asyncapi/avro-schema-parser';
+
+registerSchemaParser(openapiSchemaParser);
+registerSchemaParser(ramlDtParser);
+registerSchemaParser(avroSchemaParser);
+
+function prepareParserConfig(req?: Request) {
+  if (!req) {
+    return {
+      resolve: {
+        file: false,
+      },
+    }
+  }
+
+  return {
+    resolve: {
+      file: false,
+      http: {
+        headers: {
+          Cookie: req.header('Cookie'),
+        },
+        withCredentials: true,
```

Comment:
How this is gonna work? What are the credentials? Are they gonna be based on a cookie? I'm confused.

Reply:
It's used for parser options and it's exactly options for `json-schema-ref-parser`. More info here :) https://github.com/APIDevTools/json-schema-ref-parser/blob/main/docs/options.md#resolve-options In this same way we do in playground :)